src/elf/parser.rs

```diff
@@ -82,11 +83,13 @@ fn find_sym<'mmap>(
 struct Cache<'mmap> {
     /// A slice of the raw ELF data that we are about to parse.
     elf_data: &'mmap [u8],
-    /// The cached ELF header as well as the actual number of section headers.
+    /// The cached ELF header as well as the actual number of section
+    /// and program headers.
     ///
-    /// This latter value effectively overrides that of `ehdr.e_shnum`, which is
-    /// special-cased by the ELF standard.
-    ehdr: Option<(&'mmap Elf64_Ehdr, usize)>,
+    /// This latter two values effectively override `ehdr.e_shnum` and
+    /// `ehdr.e_phnum`, handling of which is special-cased by the ELF
+    /// standard.
+    ehdr: Option<(&'mmap Elf64_Ehdr, usize, usize)>,
```

Comment:
this starts to get confusing, why not add a small internal struct with proper field names?

Reply:
Done.